torchvision/datasets/widerface.py

```diff
@@ -0,0 +1,231 @@
+from PIL import Image
+import os
+from os.path import abspath, expanduser
+import torch
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union
+from .utils import download_file_from_google_drive, check_integrity, download_url, extract_archive
+from .vision import VisionDataset
+
+
+class WIDERFace(VisionDataset):
+    """`WIDERFace <http://shuoyang1213.me/WIDERFACE/>`_ Dataset.
+
+    Citation:
+    @inproceedings{yang2016wider,
+        author    = "Yang, Shuo and Luo, Ping and Loy, Chen Change and Tang, Xiaoou",
+        booktitle = "IEEE Conference on Computer Vision and Pattern Recognition (CVPR)",
+        title     = "WIDER FACE: A Face Detection Benchmark",
+        year      = "2016"}
```

Comment:
This would set precedence, since we do not have a citation in any other dataset. To be clear, I like the idea ("credit were credit is due"), but something like this entails a lot. @fmassa?

Reply:
I included this for two primary reasons
* to direct users to the original paper for questions about how to interpret the dataset
* there are several version of widerface annotations floating around on the internet with a different folder structure. I want to make it clear that the annotations are the original annotations released with the dataset.

If this has implications like altering the license in someway, I'm happy to leave it off. I just figured the docstring would be more helpful by directing people to the original source.